### DISCUSSION TITLE
Support for node-level proxy chain

### DISCUSSION THREAD

**Discussion by piyoki** [daeuniverse/dae#236] [Announcements]
### Scenario 1: Forwarding chains for privacy

Assume that you are A, and B is the relay server, and C is the endpoint server.

The topology is:

```
A -> B -> C
```

In this case, B in the middle can only know that A has been accessing C, and C can only know that B has been visiting various websites.

Configure dae as:

```bash
node {
  new_node_name: 'vmess://C_LINK -> trojan://B_LINK' # Encapsulate requests to C in B.
}
```

### Scenario 2: More flexible protocol by artificial

If you configured a `socks5 over wss` inbound on your v2ray server, you can use dae as the client.

Configure dae as:

```bash
node {
  # Example:
  chains2: 'socks5://yeeeah.v2raya.org:6999 -> wss://yeeeah.v2raya.org:6999/7cadf625-171c-4002-7abc-257f6485fcc2'
}
```

## Scheme & Usage

```bash
node {
  new_node_name: 'node link 1 (endpoint) -> node link 2 (relay)'
}
```

Example:

```bash
node {
  # case 1: tuic over vless (different nodes)
  chains: 'tuic://my.endpoint.com:18364/xxxxx -> vless://ohmy.relay.com:12212/xxxxx'

  # case 2: socks5 over wss (same server port, so wss is the transport layer)
  chains2: 'socks5://yeeeah.v2raya.org:6999 -> wss://yeeeah.v2raya.org:6999/7cadf625-171c-4002-7abc-257f6485fcc2'
}
```

Moreover, if you are a subscription maker, you can also use such expressions in your subscription content. And then encode them with base64:

```
tuic://my_tuic_link -> vmess://my_vmess_link
ss://my_ss_link -> vmess://my_vmess_link
```

## Implementation

- #235

**Comment by antonchen** [comment]
Does it support group-level proxy chain?

Like:

```
node {
  my_node: 'vmess://C_LINK'
  out_node: 'my_node -> proxy'
}

group {
  proxy {
    filter: name(keyword: HK, keyword: SG)
    policy: min_moving_avg
  }
}

```
